### ISSUE TITLE
yo react-native-ignite throws "'wc' is not recognized as an internal or external command, operable program or batch file" in Win10

### ISSUE THREAD

**Issue by danitk** [infinitered/ignite#91]
generate app -> AppName ☕️  This will take a while ☕️
'wc' is not recognized as an internal or external command,
operable program or batch file.
This will walk you through creating a new React Native project in C:\Users\Dani Alvarez\AppName
Installing react-native package from npm...
events.js:141
      throw er; // Unhandled 'error' event
      ^

Error: spawn npm ENOENT
    at exports._errnoException (util.js:870:11)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:178:32)
    at onErrorNT (internal/child_process.js:344:16)
    at nextTickCallbackWith2Args (node.js:442:9)
    at process._tickCallback (node.js:356:17)
    at Function.Module.runMain (module.js:443:11)
    at startup (node.js:139:18)
    at node.js:968:3
 Too many arguments.

usage: git clone [options] [--] <repo> [<dir>]

    -v, --verbose         be more verbose
    -q, --quiet           be more quiet
    --progress            force progress reporting
    -n, --no-checkout     don't create a checkout
    --bare                create a bare repository
    --mirror              create a mirror repository (implies bare)
    -l, --local           to clone from a local repository
    --no-hardlinks        don't use local hardlinks, always copy
    -s, --shared          setup as shared repository
    --recursive           initialize submodules in the clone
    --recurse-submodules  initialize submodules in the clone
    --template <template-directory>
                          directory from which templates will be used
    --reference <repo>    reference repository
    -o, --origin <name>   use <name> instead of 'origin' to track upstream
    -b, --branch <branch>
                          checkout <branch> instead of the remote's HEAD
    -u, --upload-pack <path>
                          path to git-upload-pack on the remote
    --depth <depth>       create a shallow clone of that depth
    --single-branch       clone only one branch, HEAD or --branch
    --separate-git-dir <gitdir>
                          separate git dir from working tree
    -c, --config <key=value>
                          set config inside the new repository

events.js:141
      throw er; // Unhandled 'error' event
      ^
 AssertionError: Trying to copy from a source that does not exist: C:\Users\Dani Alvarez\AppData\Roaming\npm\node_modules\generator-react-native-ignite\app\templates\ignite-base\README.md.template
    at EditionInterface.exports._copySingle (C:\Users\Dani Alvarez\AppData\Roaming\npm\node_modules\generator-react-native-ignite\node_modules\mem-fs-editor\lib\actions\copy.js:45:3)
    at EditionInterface.exports.copy (C:\Users\Dani Alvarez\AppData\Roaming\npm\node_modules\generator-react-native-ignite\node_modules\mem-fs-editor\lib\actions\copy.js:23:17)
    at EditionInterface.module.exports [as copyTpl] (C:\Users\Dani Alvarez\AppData\Roaming\npm\node_modules\generator-react-native-ignite\node_modules\mem-fs-editor\lib\actions\copy-tpl.js:12:8)
    at copyOverBase (C:\Users\Dani Alvarez\AppData\Roaming\npm\node_modules\generator-react-native-ignite\app\index.js:36:14)
    at AppGenerator.generateApp (C:\Users\Dani Alvarez\AppData\Roaming\npm\node_modules\generator-react-native-ignite\app\index.js:92:7)
    at C:\Users\Dani Alvarez\AppData\Roaming\npm\node_modules\generator-react-native-ignite\node_modules\yeoman-generator\lib\base.js:430:16
    at processImmediate [as _immediateCallback] (timers.js:383:17)

**Comment by kevinvangelder** [comment]
@danitk We haven't done any testing on Windows yet. Are you on a Windows Insider build? If not, bash commands won't work on your command line until the general release of the Windows Redstone update this summer, and even then it could be hit-and-miss.

Thanks for bringing this to our attention, but I'm not sure this is something that we'll be able to address in the near future.

**Comment by danitk** [comment]
Sorry to hear that, hopefully you guys will add Windows support in the future. I'm on Windows 10 Home edition Version 1511 Build 10586.218.

**Comment by kevinvangelder** [comment]
Can you try it from a bash prompt?

> Run native Bash on Ubuntu on Windows: In this build, you can natively run Bash in Windows as announced last week at Build 2016. To do this, you first need to turn on Developer Mode via Settings > Update & security > For developers. Then search for "Windows Features" and choose "Turn Windows features on or off" and enable Windows Subsystem for Linux (Beta). To get Bash installed, open Command Prompt and type "bash".

**Comment by GantMan** [comment]
Hey @danitk - thanks for reporting this!
The `wc` command is just part of our protection shell script.   It's not fundamental to the application running.

Can you do basic react native app initialization with `react-native init SomeApp` ?

If so, let's see about scheduling a pairing session.   @kevinvangelder - this might be a great chance to shine and get `ignite` legit with Windows.

**Comment by kevinvangelder** [comment]
@GantMan I finally got a chance to try this on my Windows 10 computer and after running `npm install --global yo bower grunt-cli` and running the generator I got this error: 
```
npm WARN optional Skipping failed optional dependency /chokidar/fsevents:
npm WARN notsup Not compatible with your operating system or architecture: fsevents@1.0.12
events.js:141
      throw er; // Unhandled 'error' event
      ^

Error: ENOENT: no such file or directory, open 'D:\Code\IgniteTest\android\app\src\main\AndroidManifest.xml'
    at Error (native)
    at Object.fs.openSync (fs.js:549:18)
    at Object.fs.readFileSync (fs.js:397:15)
    at Object.insertInFile (C:\Users\Kevin\AppData\Roaming\npm\node_modules\generator-react-native-ignite\utilities.js:21:27)
    at performInserts (C:\Users\Kevin\AppData\Roaming\npm\node_modules\generator-react-native-ignite\app\index.js:69:13)
    at AppGenerator.end (C:\Users\Kevin\AppData\Roaming\npm\node_modules\generator-react-native-ignite\app\index.js:126:7)
    at C:\Users\Kevin\AppData\Roaming\npm\node_modules\generator-react-native-ignite\node_modules\yeoman-generator\lib\base.js:430:16
    at processImmediate [as _immediateCallback] (timers.js:383:17)
```

I was then able to run `react-native android` to generate the missing Android folder, but attempting to run the app resulted in:
```
:app:preDexDebug
Unknown source file : Error: Could not create the Java Virtual Machine.
Unknown source file : Error: A fatal exception has occurred. Program will exit.

:app:preDexDebug FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:preDexDebug'.
> com.android.ide.common.process.ProcessException: org.gradle.process.internal.ExecException: Process 'command 'C:\Program Files\Java\jdk1.7.0_79\bin\java.exe'' finished with non-zero exit value 1

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.

BUILD FAILED

Total time: 17.426 secs
Could not install the app on the device, read the error above for details.
Make sure you have an Android emulator running or a device connected and have
set up your Android development environment:
https://facebook.github.io/react-native/docs/android-setup.html
```
I think I encountered this same error when writing my blog post, I just don't remember how I fixed it off the top of my head, so I'll have to revisit that to see if I can get it working.

TL;DR: Didn't fix anything, but got some additional information we may be able to use to track down the issues.

**Comment by kevinvangelder** [comment]
Interestingly, running `ignite new MyProject` on OSX automatically detects that yoeman is not installed and installs it where it didn't do that when I was testing on Windows yesterday.

**Comment by davza** [comment]
After running:
npm install --global yo bower grunt-cli
then
npm install --global react-native-ignite
then
ignite new MyTestIgniteApp

The app was generated properly.
I had to open the /android folder up in Android Studio as with any react-native project and after that the following worked fine for running the Ignite generated application in the Android emulator using:
react-native run-android

This was on a standard Windows 10 Pro system (1511, build: 10586.420)

**Comment by GantMan** [comment]
@davza Very nice!   I'm eager to get Ignite windows friendly now!

Were there any other hiccups?

**Comment by GantMan** [comment]
closing in favor of https://github.com/infinitered/ignite/issues/196

**Comment by davza** [comment]
@GantMan No problems so far.
I generated a signed apk of the base ignite project and it installed fine on my phone.
I'm going to be moving parts my current react-native app into ignite to see how it works and will let you know if I have any problems.

Thanks for the tools :)
